File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.py
from collections import deque

s, y = map(int, input().split())
arr = [-1] * 100001  # 방문 여부를 -1로 초기화

def bfs(x):
    if x == y:  # 시작점이 도착점과 같으면 바로 0 반환
        return 0

    arr[x] = 0  # 시작점은 0으로 설정
    q = deque([x])

    while q:
        x = q.popleft()

        # 현재 위치에서 가능한 다음 이동 계산
        cal = [x - 1, x + 1, 2 * x]

        for i in cal:
            if 0 <= i <= 100000 and arr[i] == -1:  # 방문하지 않은 곳만
                arr[i] = arr[x] + 1
                if i == y:  # 목표 지점에 도달하면 결과 반환
                    return arr[i]
                q.append(i)

# 결과 출력
print(bfs(s))

File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "dev/stdin" : "././input.txt";
const [s, y] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split(" ")
  .map(Number);

const arr = new Array(100001).fill(-1);

if (s === y) console.log(0);
else {
  arr[s] = 0;

  const queue = [s];
  let head = 0;

  while (queue.length > 0) {
    const x = queue[head++];

    for (const nx of [x - 1, x + 1, x * 2]) {
      if (nx < 0 || nx > 100000 || arr[nx] !== -1) continue;

      arr[nx] = arr[x] + 1;

      if (nx === y) {
        console.log(arr[nx]);
        process.exit(0);
      }

      queue.push(nx);
    }
  }
}
